File: android/app/src/main/kotlin/com/example/test_twilio/arguments/ChannelArgument.kt
package com.example.test_twilio.arguments

class ChannelArgument(
        var friendlyName: String?,
        var sid: String?,
        var dateUpdatedAsDate: Long?,
        var dateCreatedAsDate: Long?,
        var status: Int?,
        var lastMessageDate: Long?,
        var notificationLevel: Int?,
        var lastMessageIndex: Int?,
        var type: Int?) {

    companion object {
        fun fromMap(data: HashMap<String, Any>?): ChannelArgument {
            return ChannelArgument(
                    friendlyName = data?.get("friendlyName")?.toString(),
                    sid = data?.get("sid")?.toString(),
                    dateUpdatedAsDate = data?.get("dateUpdatedAsDate")?.toString()?.toLong(),
                    dateCreatedAsDate = data?.get("dateCreatedAsDate")?.toString()?.toLong(),
                    status = data?.get("status")?.toString()?.toInt(),
                    lastMessageDate = data?.get("lastMessageDate")?.toString()?.toLong(),
                    notificationLevel = data?.get("notificationLevel")?.toString()?.toInt(),
                    lastMessageIndex = data?.get("lastMessageIndex")?.toString()?.toInt(),
                    type = data?.get("type")?.toString()?.toInt()
            )
        }
    }
}

File: ios/Runner/arguments/ChannelArgument.swift
//
//  ChannelArgument.swift
//  Runner
//
//  Created by KienNT on 10/26/21.
//

import Foundation

class ChannelArgument {
    var friendlyName: String?
    var sid: String?
    var dateUpdatedAsDate: Int64?
    var dateCreatedAsDate: Int64?
    var status: Int?
    var lastMessageDate: Int64?
    var notificationLevel: Int?
    var lastMessageIndex: Int?
    var type: Int?
    
    init(friendlyName: String?, sid: String?, dateUpdatedAsDate: Int64?, dateCreatedAsDate: Int64?, status: Int?, lastMessageDate: Int64?, notificationLevel: Int?, lastMessageIndex: Int?, type: Int?) {
        self.friendlyName = friendlyName
        self.sid = sid
        self.dateUpdatedAsDate = dateUpdatedAsDate
        self.dateCreatedAsDate = dateCreatedAsDate
        self.status = status
        self.lastMessageDate = lastMessageDate
        self.notificationLevel = notificationLevel
        self.lastMessageIndex = lastMessageIndex
        self.type = type
    }
    
    static func fromMap(data: [String: Any?]) -> ChannelArgument {
        return ChannelArgument(friendlyName: data["friendlyName"] as? String, sid: data["sid"] as? String, dateUpdatedAsDate: data["dateUpdatedAsDate"] as? Int64, dateCreatedAsDate: data["dateCreatedAsDate"] as? Int64, status: data["status"] as? Int, lastMessageDate: data["lastMessageDate"] as? Int64, notificationLevel: data["notificationLevel"] as? Int, lastMessageIndex: data["lastMessageIndex"] as? Int, type: data["type"] as? Int)
    }
}
